_posts/2025-06-04-golang-template-with-opentelemetry-support.md

```diff
@@ -0,0 +1,430 @@
+---
+title: "Introducing a New OpenFaaS Golang Template with OpenTelemetry Support"
+description: "Explore the new OpenFaaS Golang template with built-in OpenTelemetry for better observability."
+date: 2025-06-04
+author_staff_member: han
+categories:
+  - golang
+  - opentelemetry
+  - template
+dark_background: true
+image: ""
+hide_header_image: true
+---
+
+We have created a new OpenFaaS template for Golang with OpenTelemetry support.
+
+In our previous blog post on OpenTelemetry, [How to Trace functions with OpenTelemetry](https://www.openfaas.com/blog/trace-functions-with-opentelemetry/), we showed how [auto-instrumentation](https://opentelemetry.io/docs/zero-code/) can be used to quickly instrument Python and Node.js functions. In this post, you’ll learn how to add tracing to your Golang functions in OpenFaaS using a new template we’ve created with baked-in OpenTelemetry support. We’ll walk through some examples, including querying a PostgreSQL database and adding custom spans.
+
+![Grafana visualisation for the default HTTP trace from a function using the golang-otel template.](/images/2025-05-opentelemetry-golang/http-trace.png)
+> Grafana visualisation for the default HTTP trace from a function using the golang-otel template. 
+
+[OpenTelemetry](https://opentelemetry.io/) is an open-source observability framework that provides a standardized way to instrument, generate, collect, and export telemetry data like traces from your applications. Traces specifically represent the end-to-end journey of a single request as it flows through various services and components within a distributed system, allowing you to visualize its path and identify performance bottlenecks or errors.
+
+Configuring an exporter manually and adding traces to existing Golang functions requires quite some boilerplate code. While there is support for auto-instrumentation for Go, the [OpenTelemetry Go Auto-instrumentation project](https://github.com/open-telemetry/opentelemetry-go-instrumentation) is still in beta. Moreover, it requires a privileged container which we strongly discourage if possible. 
```

Comment:
"Without this template, configuring an exporter manually...."

Adding a prefix tries to avoid the reader prematurely misunderstanding